sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTextBasedPartitionReader.scala

```diff
@@ -212,6 +207,24 @@ abstract class GpuTextBasedPartitionReader(
    */
   def getFileFormatShortName: String
 
+  /**
+   * Handle the table decoded by GPU
+   * @param readDataSchema the Spark schema describing what will be read
+   * @param table the table decoded by GPU
+   * @return the new optional Table
+   */
+  def handleResult(readDataSchema: StructType, table: Table): Option[Table] = {
+    val numColumns = table.getNumberOfColumns
+
+    closeOnExcept(table) { _ =>
```

Comment:
This is odd.  Normally there are two ways to handle a GPU resource being passed to a method:
1. The method owns it and is responsible for closing the GPU resource, exceptions or not.
2. The caller owns it and is responsible for closing it, exceptions or not.

However this method closes it only when there's an exception but otherwise does not.  For this to work properly, that implies the caller is not protecting the GPU resource it owns from an exception.  If it is properly doing that, the caller will end up double-closing the GPU resource because this will have closed it first.  IMO this method should not own the table and rely on the caller to properly handle exceptions, from here or anywhere else while it is responsible for the table.

Reply:
This case should match the scenario `The method owns it and is responsible for closing the GPU resource, exceptions or not.`

For the GPU resource handling convention, we should close the input table and return a new
table just like the below code. But considering the code simplicity and optimization, we just return the input table.

withResource(table) { _
   val cols = (0 until  table.getNumberOfColumns).map(i => table.getColumn(i))
  Some(new Table(cols: _*))
}

Reply:
Added some comments in the code.